scipy/stats/_distn_infrastructure.py

```diff
@@ -3889,6 +3889,8 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
             raise ValueError("All elements of pk must be non-negative.")
         if not np.allclose(np.sum(pk), 1):
             raise ValueError("The sum of provided pk is not 1.")
+        if not len(set(np.ravel(xk))) == np.size(xk):
+            raise ValueError("xk may not contain duplicate values.")
```

Comment:
If we were doing this today, it would be more efficient to turn `xk` into an array from the outset then call the methods `ravel` and `size` so that it does not need to be converted into an array by NumPy each time. But again, this is consistent with what it is here, and I don't think we need another CI run.